model/project_parser.go

```diff
@@ -630,6 +631,58 @@ func GetProjectFromBSON(data []byte) (*Project, error) {
 	return TranslateProject(pp)
 }
 
+func processIntermediateProjectInludes(ctx context.Context, identifier string, intermediateProject *ParserProject,
+	include Include, yamlChan chan<- yamlTuple, projectOpts *GetProjectOpts) {
+	// Make a copy of opts because otherwise parts of opts would be
+	// modified concurrently.  Note, however, that Ref and PatchOpts are
+	// themselves pointers, so should not be modified.
+	localOpts := &GetProjectOpts{
+		Ref:             projectOpts.Ref,
+		PatchOpts:       projectOpts.PatchOpts,
+		LocalModules:    projectOpts.LocalModules,
+		RemotePath:      include.FileName,
+		Revision:        projectOpts.Revision,
+		Token:           projectOpts.Token,
+		ReadFileFrom:    projectOpts.ReadFileFrom,
+		Identifier:      identifier,
+		UnmarshalStrict: projectOpts.UnmarshalStrict,
+	}
+	localOpts.UpdateReadFileFrom(include.FileName)
+
+	var yaml []byte
+	var err error
+	grip.Debug(message.Fields{
+		"message":     "retrieving included YAML file",
+		"remote_path": localOpts.RemotePath,
+		"read_from":   localOpts.ReadFileFrom,
+		"module":      include.Module,
+	})
+	if include.Module != "" {
+		yaml, err = retrieveFileForModule(ctx, *localOpts, intermediateProject.Modules, include.Module)
+	} else {
+		yaml, err = retrieveFile(ctx, *localOpts)
+	}
+	if err != nil {
+		yamlChan <- yamlTuple{
+			yaml: nil,
+			name: include.FileName,
+			err:  errors.Wrapf(err, "%s: retrieving file '%s'", LoadProjectError, include.FileName),
```

Comment:
I'd refer if this error was a bit different for 660 / 662 so that we could better trace the path, if there is an error. Maybe we can set the wrapped error within the if/else so we can add a "with modules"